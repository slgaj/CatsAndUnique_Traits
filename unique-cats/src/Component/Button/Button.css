.catHome {
    background-repeat:no-repeat;      
    background-size:100%;
    background-position-y: bottom;    
    height: 143px;    
    cursor: pointer;
    border: none;
    padding: 0 20px;
    margin:20px;    
}

.square {
    background: #fff;
    border: 1px solid #999;
    float: left;
    font-size: 24px;
    font-weight: bold;
    line-height: 34px;
    height: 34px;
    margin-right: -1px;
    margin-top: -1px;
    padding: 0;
    text-align: center;
    width: 34px;
}

.square:focus {
    outline: none;
}

.reset-btn {
    margin-top: 20px;
    padding: 10px 20px;
    background: #fff;
    border: 2px solid #690e63;
    border-radius: 10px;
    cursor: pointer;
    outline: none;
}