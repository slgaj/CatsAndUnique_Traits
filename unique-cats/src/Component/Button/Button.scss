@import '../../assets/GlobalCss/global.scss';

.catHome {
    background-repeat: $background-repeat-none;      
    background-size: 100%;
    background-position-y: bottom;    
    height: 143px;    
    cursor: $pointer;
    border: $border-zero;
    padding: 0 $p20;
    margin: $m20; 
    
    @media (min-width:1024px) {
             
        background-size: 56%;
        height: 104px;
        width: 100px;         
        margin: 5px $m10;
        background-position: $pos-center;   
    
    }
}

.reset-btn {
    margin-top: $m20;
    padding: $p10 $p20;
    background: $serene-white;
    border: 2px solid #690e63;
    border-radius: 10px;
    cursor: $pointer;
    outline: $border-zero;
}