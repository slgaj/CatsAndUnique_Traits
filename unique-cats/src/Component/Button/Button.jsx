import React from "react";
import "./Button.css";

export default function Button(props) {
  return (
    <div
      className={props.class}
      value={props.value}
      style={{
        backgroundImage: `url(${props.imageName})`,
        minWidth: `${props.size}px`,
        backgroundSize: `${props.bgSize}%`
      }}
      onClick={props.onClick}
    >
      {props.children}
    </div>
  );
}

