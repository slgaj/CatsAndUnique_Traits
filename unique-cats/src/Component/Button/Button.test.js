import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Button from './Button';

import '../../setupTests';

const fnClick = jest.fn();
describe("MyComponent", () => {
  it("should render my component", () => {
    const wrapper = shallow(<Button />);
  });
});

it('matches the snapshot', () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('click events', () => {
    it('button click should add Cat to Grid', () => {
        //replace actual function with mock function
        const component = shallow(<Button class={"catHome"} onClick={fnClick} />);
        
        //simulate a click
        component
        .find('div.catHome')
        .simulate('click');
        
        //check if function was called
        expect(fnClick).toHaveBeenCalled();
    });
    });


