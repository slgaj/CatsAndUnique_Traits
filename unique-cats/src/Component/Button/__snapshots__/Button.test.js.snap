// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<div
  style={
    Object {
      "backgroundImage": "url(undefined)",
      "backgroundSize": "undefined%",
      "minWidth": "undefinedpx",
    }
  }
/>
`;
